pkg/event/target/mqtt.go

```diff
@@ -104,7 +108,12 @@ func (target *MQTTTarget) Send(eventData event.Event) error {
 		return err
 	}
 
-	token := target.client.Publish(target.args.Topic, target.args.QoS, false, string(data))
+	var token mqtt.Token
+	if !target.client.IsConnected() {
+		token = target.client.PublishNonRegisteredEvent(target.args.Topic, target.args.QoS, false, string(data))
```

Comment:
How about keeping this logic transparent in `target.client.Publish()` than introducing new method?

Reply:
Done :+1: